### ISSUE TITLE
-DCMAKE_INSTALL_PREFIX has no effect when building with CMAKE

### ISSUE THREAD

**Issue by mrawding** [tensorflow/tensorflow#49306]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution = Linux Ubuntu 20.04:
- TensorFlow installed from (source or binary): CMake efficient binary generation
- TensorFlow version: latest
- Python version: python 3.8
- Installed using virtualenv? pip? conda?:
- Bazel version (if compiling from source): n/a
- GCC/Compiler version (if compiling from source): 9.3.0
- CUDA/cuDNN version: 11.2
- GPU model and memory: Quadro RTX 4000 - 8GB

When following the instructions to build Tensorflow Lite as a cmake package with libraries and header files, it does not install anything to the install prefix. I followed the instructions here:

https://www.tensorflow.org/lite/guide/build_cmake 

Is there a command I am missing. After successful configuration and build, I try to run:

```
cmake --install . 

```

All that happens is the printout statement:
```
-- Install configuration: "Release"
```

I am trying to install tflite as a cmake package that other projects can use the library and header files. However, it needs to be in a specific install location. Currently I am creating a Cmake project that uses "ExternalProject" to clone, configure, build, and install tensorflowlite. The only issue is that it does not install to the location I need it to even when specifying -DCMAKE_INSTALL_PREFIX.

Previously I used bazel to build from source, copy all the files, and also generate a cmake package configuration. However, I want to be able to build the entire package with just CMakelists.txt. I would appreciate any help on how to have the libraries and include files installed into a single location throuh Cmake. 

Thank you so much! I am sure a lot of users would really like this feature, especially in the robotics community.

**Comment by tilakrayal** [comment]
@mrawding ,
Is this still an issue?
Could you please update TensorFlow to the latest stable version v.2.6 and let us know if you are facing the same error. Thanks!

**Comment by mrawding** [comment]
@tilakrayal yes this is still an issue. All the includes and libraries are installed to the build directory, but you still need to search and copy them to another directory to package them. The initial ```cmake`` command does recognize the option for Eigen, but still no libraries are installed. Here are the steps I ran:

1. Clone tensorflow
2. ```mkdir build```
3. ```mkdir install_space```
4. ```cd build```
5. ``` cmake ../tensorflow/tensorflow/lite -DCMAKE_INSTALL_PREFIX=/home/mike/install_space``` (successful)
6. ``` cmake --build  . -j$(nproc) (successful)
7. ``` cmake --install . --prefix /home/mike/install_space ````
Outputs: ```-- Install configuration: "Release" ```

No files installed in /home/mike/install_space

**Comment by Saduf2019** [comment]
@mrawding 
Could you please try on the latest tf version as many bugs have been fixed.

**Comment by tenta4** [comment]
Issue persists on v2.9.0

**Comment by brightening-eyes** [comment]
It still persists on 2.10
cmake does not install it at all

**Comment by mohantym** [comment]
Hi @mrawding !

I could replicate this issue in 2.10 . Prefix is not working as @brightening-eyes ! I pointed out. 

Thank you!

**Comment by pjpratik** [comment]
Hi @mrawding 

I have pulled the recent master branch and tried to reproduce locally with the instructions given

> 1. Clone tensorflow
> 2. mkdir build
> 3. mkdir install_space
> 4. cd build
> 5.  cmake ../tensorflow/tensorflow/lite -DCMAKE_INSTALL_PREFIX=/home/mike/install_space (successful)
> 6. cmake --build  . -j$(nproc)  (successful)
> 7. cmake --install . --prefix /home/mike/install_space

and was able to successfully install with the prefix. Please find the screenshot below

<img width="1580" alt="image" src="https://github.com/tensorflow/tensorflow/assets/118897289/a3fcd879-ded4-4615-a7e2-6c3c11457278">

Thanks.

**Comment by tenta4** [comment]
For v2.13.0 I can confirm something was installed, but seems there are only 3rd-patries
```> du -sh install/*
3.9M    install/bin
13M     install/include
3.3M    install/lib
40K     install/share

> ls -l install/include/
total 1136
-rw-r--r-- 1 tenta4 tenta4  48401 Aug  1 18:24 cpuinfo.h
drwxr-xr-x 4 tenta4 tenta4   4096 Aug  1 18:24 eigen3
drwxr-xr-x 3 tenta4 tenta4   4096 Aug  1 18:24 flatbuffers
drwxr-xr-x 2 tenta4 tenta4   4096 Aug  1 18:24 fp16
-rw-r--r-- 1 tenta4 tenta4    141 Aug  1 18:24 fp16.h
-rw-r--r-- 1 tenta4 tenta4  13149 Aug  1 18:24 fxdiv.h
-rw-r--r-- 1 tenta4 tenta4 811158 Aug  1 18:24 NEON_2_SSE.h
-rw-r--r-- 1 tenta4 tenta4  45504 Aug  1 18:24 psimd.h
-rw-r--r-- 1 tenta4 tenta4  47964 Aug  1 18:24 pthreadpool.h
-rw-r--r-- 1 tenta4 tenta4 166749 Aug  1 18:24 xnnpack.h

**Comment by pjpratik** [comment]
Hi @tenta4 

Can you please create a new issue to track it separately since the original issue is resolved on v2.13?

Thanks.
